pdns/rec_channel_rec.cc

```diff
@@ -1864,9 +1908,25 @@ static string setEventTracing(T begin, T end)
   }
 }
 
-static void* pleaseSupplantProxyMapping(std::shared_ptr<ProxyMapping> pm)
+static void* pleaseSupplantProxyMapping(const ProxyMapping& pm)
 {
-  t_proxyMapping = pm;
+  if (pm.empty()) {
+    t_proxyMapping = nullptr;
+  }
+  else {
+    // Copy the existing stats values, for the new config items also present in the old
+    auto newmapping = make_unique<ProxyMapping>();
+    for (const auto& [nm, entry] : pm) {
+      auto& newentry = newmapping->insert(nm);
+      newentry.second = entry;
+      if (t_proxyMapping) {
+        if (auto existing = t_proxyMapping->lookup(nm); existing != nullptr) {
```

Comment:
Clang-tidy complains about this line, `auto existing` can be changed to `const auto *existing`.